network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +340,25 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `MAX_FULLNODE_CONNECTIONS`
+        // including in flight dials.
+        let to_connect_size = match self.network_context.role() {
+            RoleType::FullNode => min(
+                to_connect.len(),
+                MAX_FULLNODE_CONNECTIONS - self.connected.len() - self.dial_queue.len(),
+            ),
+            _ => to_connect.len(),
+        };
```

Comment:
this feels loong, I wonder if an if let else would only be 1 line less, I guess :/

Reply:
Ah, it should be an `if let`

Reply:
:( I actually doesn't get shorter with an if else, because of the else clause.